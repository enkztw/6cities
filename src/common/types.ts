export type property = `apartment` | `room` | `house` | `hotel`;
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`
export type sort = `rating` | `id` | `toLow` | `toHigh`


