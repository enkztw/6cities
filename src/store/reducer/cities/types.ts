enum Types {
  SET_CITY = `SET_CITY`
}

export default Types;
