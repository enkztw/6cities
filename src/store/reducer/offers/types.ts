enum Types {
  SET_OFFER = `SET_OFFER`,
  SET_COMMENTS = `SET_COMMENTS`,
  SET_OFFERS = `SET_OFFERS`,
  SET_FAVORITES = `SET_FAVORITES`,
  SET_NEARBY = `SET_NEARBY`,
  SET_SORTING = `SET_SORTING`
}

export default Types;
