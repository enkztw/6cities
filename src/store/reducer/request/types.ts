enum Types {
  SET_REQUEST = `SET_REQUEST`,
}

export default Types;

