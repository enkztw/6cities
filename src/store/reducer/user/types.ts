enum Types {
  SET_AUTH = `SET_AUTH`
}

export default Types;

