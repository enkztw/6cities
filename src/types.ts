// import OperationStatus from './const';

export type property = `apartment` | `room` | `house` | `hotel`;
// export type status = OperationStatus.SUCCESS |
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`


