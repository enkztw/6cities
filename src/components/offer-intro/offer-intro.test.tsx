import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Offer} from '../../types';
import Intro from './offer-intro';

const offer: Offer = {
  id: 1,
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    }
  },
  previewImage: `img/apartment-01.jpg`,
  images: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: false,
  rating: 1.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 100,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    id: 3,
    isPro: true,
    name: `Angelina`,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  }
};

describe(`Render <Intro />`, () => {
  it(`<Intro /> should render`, () => {
    const tree = renderer.create(<Intro {...offer} errorStatus={false} onClick={jest.fn()}/>);

    expect(tree).toMatchSnapshot();
  });

  it(`<Intro /> should render with premium, book marks & error`, () => {
    const tree = renderer.create(<Intro {...offer} isPremium={true} isFavorite={true} errorStatus={true} onClick={jest.fn()}/>);

    expect(tree).toMatchSnapshot();
  });
});
