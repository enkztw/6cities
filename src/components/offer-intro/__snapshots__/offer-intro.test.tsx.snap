// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Intro /> <Intro /> should render 1`] = `
Array [
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Beautiful & luxurious studio at great location
    </h1>
    <button
      className="property__bookmark-button button"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      1.8
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      3
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      100
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
]
`;

exports[`Render <Intro /> <Intro /> should render with premium and book marks 1`] = `
Array [
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>,
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Beautiful & luxurious studio at great location
    </h1>
    <button
      className="property__bookmark-button button property__bookmark-button--active"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      1.8
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      3
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      100
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
]
`;
